### ISSUE TITLE
"LR1 TLS not installed"

### ISSUE THREAD

**Issue by willcrichton** [lalrpop/lalrpop#145]
With the following grammar:

```
grammar;
pub A: () = { B+ => () };
B: () = { "c"+ => () };
```

I get "LR1 TLS not installed." I'm not sure if it's saying this because the grammar is ambiguous, but if so, it would be nice to have a less opaque error message.

**Comment by nikomatsakis** [comment]
I think I fixed this, but maybe only in the trunk release. Try switching cargo to use the github master branch...I will hopefully issue a new release soon.

**Comment by brendanzab** [comment]
Yeah - still getting this. It's waiting on the release.

**Comment by yberreby** [comment]
I just hit it with the following grammar:

```rust
use std::str::FromStr;
use ast::{Operator, Expr, Operation};

// Grammar:

//    number : /-?[0-9]+/ ;
//    symbol : '+' | '-' | '*' | '/' ;
//    sexpr  : '(' <expr>* ')' ;
//    expr   : <number> | <symbol> | <sexpr> ;
//    lispy  : /^/ <expr>* /$/ ;
//

grammar;

pub SExpr: Vec<Expr> = <Expr*>;

Expr: Expr = {
  <i:Integer> => Expr::Integer(i),
  <i:Symbol> => Expr::Symbol(i),
  <sexpr:SExpr> => Expr::SExpr(sexpr),
};

Integer: i64 = {
  <s:r"-?[0-9]+"> => i64::from_str(s).expect("failed to parse integer"),
};

Symbol: String = r"[a-zA-Z_+*\\-\\/\\\\=<>!&]+";
```

**Comment by nikomatsakis** [comment]
I'll issue a new release shortly -- I just want to fix #144 first (hopefully have that very soon).

**Comment by ahmedcharles** [comment]
This now gives (with 0.14):

```
x.lalrpop:3:11: 3:21: Local ambiguity detected

  The problem arises after having observed the following symbols in the input:
    B+ "c"+
  At that point, if the next token is a `"c"`, then the parser can proceed in
  two different ways.

  First, the parser could execute the production at x.lalrpop:3:11: 3:21, which
  would consume the top 1 token(s) from the stack and produce a `B`. This might
  then yield a parse tree like
    "c"+ ╷ B
    ├─B──┤ │
    ├─B+─┘ │
    └─B+───┘

  Alternatively, the parser could shift the `"c"` token and later use it to
  construct a `"c"+`. This might then yield a parse tree like
    B+ "c"+ "c"
    │  ├─"c"+─┤
    │  └─B────┤
    └─B+──────┘

  See the LALRPOP manual for advice on making your grammar LR(1).

x.lalrpop:2:15: 2:23: Conflict detected

    when in this state:
  "c"+ = (*) "c"+ "c" ["c", EOF]
  "c"+ = (*) "c" ["c", EOF]
  A = B+ (*) ["c", EOF]
  B = (*) "c"+ ["c", EOF]
  B+ = B+ (*) B ["c", EOF]

  and looking at a token `"c"` we can reduce to a `A` but we can also shift

x.lalrpop:3:11: 3:21: Local ambiguity detected

  The problem arises after having observed the following symbols in the input:
    B+ "c"+
  At that point, if the next token is a `"c"`, then the parser can proceed in
  two different ways.

  First, the parser could execute the production at x.lalrpop:3:11: 3:21, which
  would consume the top 1 token(s) from the stack and produce a `B`. This might
  then yield a parse tree like
    "c"+ ╷ B
    ├─B──┤ │
    ├─B+─┘ │
    └─B+───┘

  Alternatively, the parser could shift the `"c"` token and later use it to
  construct a `"c"+`. This might then yield a parse tree like
    B+ "c"+ "c"
    │  ├─"c"+─┤
    │  └─B────┤
    └─B+──────┘

  See the LALRPOP manual for advice on making your grammar LR(1).
```
